MVP for Indian Stocks in qlib using yahooquery

2796gaurav opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
Added NSE stock data option using yahooquery.

## Description
<!--- Describe your changes in detail -->
created new class -> YahooCollectorIN1d, YahooCollectorIN
and function get_in_stock_symbols() to get NSE stock symbols which works for Indian stock market.
It doesnt conflict with existing code and if we really need to support Indian stock data, we would have to change in workflow and interface layer. Currently i have completed for data layer.

you can check to download the data from below commands

Get data:
`python  scripts/data_collector/yahoo/collector.py download_data --source_dir ~/.qlib/stock_data/source/in_1min --region IN --start 2020-11-01 --end 2020-11-10 --delay 0.1 --interval 1d`

Convert to .bin
`python scripts/dump_bin.py dump_all --csv_path ~/.qlib/stock_data/source/in_1min --qlib_dir ~/.qlib/qlib_data/my_data --include_fields open,close,high,low,volume,factor`


## Motivation and Context
<!--- Are there any related issues? If so, please put the link here. -->
No this is a feature request
<!--- Why is this change required? What problem does it solve? -->
It will provide more reach and capability for Indian stock data.

## How Has This Been Tested?
- [ ] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [ ] If you are adding a new feature, test on your own test scripts.

<!--- **ATTENTION**: If you are adding a new feature, please make sure your codes are **correctly tested**. If our test scripts do not cover your cases, please provide your own test scripts under the `tests` folder and test them. More information about test scripts can be found [here](https://docs.python.org/3/library/unittest.html#basic-example), or you could refer to those we provide under the `tests` folder. -->

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Fix bugs
- [x] Add new feature
- [ ] Update documentation

2796gaurav commented:
Hi It looks like the code was not formated by black previously. When i applied it have reformatted many other functions too.

zhupr commented:
@2796gaurav  Hi, 
Thanks for the PR, `black` add the `-l` parameter: `black qlib -l 120`

zhupr commented:
Need to add `YahooNormalizeIN`(*get the trading calendar for Indian stocks*) and `YahooNormalizeIN1d`, ref: [YahooNormalizeUS](https://github.com/microsoft/qlib/blob/main/scripts/data_collector/yahoo/collector.py#L689)    [YahooNormalizeUS1d](https://github.com/microsoft/qlib/blob/main/scripts/data_collector/yahoo/collector.py#L695)

2796gaurav commented:
Hi @zhupr I have added the calender for indian stocks, however i didnt quite understand fully how this would work?

zhupr commented:
@2796gaurav Hi, 
Needed when normalizing stock data([collector and normalize](https://github.com/microsoft/qlib/tree/main/scripts/data_collector/yahoo#collector-yahoofinance-data-to-qlib) : `python scripts/data_collector/yahoo/collector.py normalize_data`);
 normalize requires aligning each stock to a "standard trading calendar" and then doing some calculations (calculating adjusted price: [normalize yahoo data](https://github.com/microsoft/qlib/blob/main/scripts/data_collector/yahoo/collector.py#L353), etc.)

zhupr commented:
@2796gaurav You can add "region `IN`" the document: [Collector YahooFinance data to qlib](https://github.com/microsoft/qlib/tree/main/scripts/data_collector/yahoo#collector-yahoofinance-data-to-qlib)

2796gaurav commented:
Hi @zhupr Updated readme ad mentioned.
For Normalization, added function `get_calendar_list("IN_ALL")` to get Indian Normalized data
also added code for 1min data

you-n-g commented:
We've fixed a CI error recently https://github.com/microsoft/qlib/commit/509bfcb02e694efad7aff5e9df9bcafdd0de814d
Please rebase the PR to the main branch or merge the main branch to pass the CI test. 

Thanks

2796gaurav commented:
Thanks 👍
